spec-dtslint/operators/scan-spec.ts

```diff
@@ -0,0 +1,19 @@
+import { of } from 'rxjs';
+import { scan } from 'rxjs/operators';
+
+it('should enforce parameter', () => {
+  const a = of(1, 2, 3).pipe(scan()); // $ExpectError
+});
+
+it('should infer correctly ', () => {
+  const a = of(1, 2, 3).pipe(scan((x, y, z) => x + 1)); // $ExpectType Observable<number>
+});
+
+it('should infer correctly for accumulator of type array', () => {
+  const a = of(1, 2, 3).pipe(scan((x: number[], y: number, i: number) => x, [])); // $ExpectType Observable<number[]>
```

Comment:
Hi @cartant  ,
I have changed the test to identically like reduce. But, I'm not sure about the type expectations you mentioned on the comment above - look like the typings are exactly the same as reduce.
Is this correct?